core/src/main/java/org/pentaho/platform/dataaccess/datasource/wizard/csv/CsvUtils.java

```diff
@@ -462,21 +454,42 @@ public ModelInfo getModelInfo( String project, String filename ) throws FileNotF
     return (ModelInfo) xstream.fromXML( fis );
   }
 
-  private DataType convertDataType( int type ) {
+  protected int convertPrecision(  ValueMetaInterface meta  ) {
+    return meta.getPrecision() > 0 ? meta.getPrecision() : 0;
+  }
+
+  protected DataType convertDataType(  ValueMetaInterface meta  ) {
+    return convertDataType( meta.getType() );
+  }
+
+  protected DataType convertDataType( int type ) {
     switch ( type ) {
-      case 1:
-      case 5:
-      case 6:
+      case ValueMetaInterface.TYPE_NUMBER:
+      case ValueMetaInterface.TYPE_INTEGER:
+      case ValueMetaInterface.TYPE_BIGNUMBER:
         return DataType.NUMERIC;
-      case 3:
+      case ValueMetaInterface.TYPE_DATE:
         return DataType.DATE;
-      case 4:
+      case ValueMetaInterface.TYPE_BOOLEAN:
         return DataType.BOOLEAN;
       default:
         return DataType.STRING;
     }
   }
 
+  protected int convertLength( ValueMetaInterface meta ) {
+    int size;
+    if ( meta.isString() ) {
+      // pad the string lengths
+      size = meta.getLength() + ( meta.getLength() / 2 );
+    } else if ( meta.isInteger() ) {
+      size = meta.getLength();
+    } else {
+      size = meta.getPrecision() > 0 ? meta.getLength() : 0;
```

Comment:
Logic seems legit, this is handling an edge case. Not going to second guess implementation.